src/modules/battery_status/analog_battery.cpp

```diff
@@ -0,0 +1,145 @@
+	_analog_param_handles.a_per_v_old = param_find("BAT_A_PER_V");
+	_analog_param_handles.adc_channel_old = param_find("BAT_ADC_CHANNEL");
+
+	updateParams(true);
+}
+
+void
+AnalogBattery::updateBatteryStatusRawADC(hrt_abstime timestamp, int32_t voltage_raw, int32_t current_raw,
+		bool selected_source, int priority, float throttle_normalized,
+		bool armed)
+{
+	updateParams(false);
+
+	float voltage_v = (voltage_raw * _analog_params.cnt_v_volt) * _analog_params.v_div;
+	float current_a = ((current_raw * _analog_params.cnt_v_curr) - _analog_params.v_offs_cur) * _analog_params.a_per_v;
+
+	bool connected = voltage_v > BOARD_ADC_OPEN_CIRCUIT_V &&
+			 (BOARD_ADC_OPEN_CIRCUIT_V <= BOARD_VALID_UV || is_valid());
+
+
+	Battery::updateBatteryStatus(timestamp, voltage_v, current_a, connected,
+				     selected_source, priority, throttle_normalized, armed, _params.source == 0);
+}
+
+/**
+ * Whether the ADC channel for the voltage of this battery is valid.
+ * Corresponds to BOARD_BRICK_VALID_LIST
+ */
+bool AnalogBattery::is_valid()
+{
+#ifdef BOARD_BRICK_VALID_LIST
+	bool valid[BOARD_NUMBER_BRICKS] = BOARD_BRICK_VALID_LIST;
+	return valid[_index - 1];
+#else
+	// TODO: Maybe return false instead?
+	return true;
+#endif
+}
+
+int AnalogBattery::getVoltageChannel()
+{
+	if (_analog_params.adc_channel >= 0) {
+		return _analog_params.adc_channel;
+
+	} else {
+		return DEFAULT_V_CHANNEL[_index];
+	}
+}
+
+int AnalogBattery::getCurrentChannel()
+{
+	// TODO: Possibly implement parameter for current sense channel
+	return DEFAULT_I_CHANNEL[_index];
+}
+
+bool
+AnalogBattery::updateParams(bool force)
+{
+	if (Battery::updateParams(force)) {
+
+		if (_index == 1) {
+			migrateParam<float>(_analog_param_handles.v_div_old, _analog_param_handles.v_div, &_analog_params.v_div_old,
+					    &_analog_params.v_div, force);
+			migrateParam<float>(_analog_param_handles.a_per_v_old, _analog_param_handles.a_per_v, &_analog_params.a_per_v_old,
+					    &_analog_params.a_per_v, force);
+			migrateParam<int>(_analog_param_handles.adc_channel_old, _analog_param_handles.adc_channel,
+					  &_analog_params.adc_channel_old, &_analog_params.adc_channel, force);
+
+			_param_sub.update(&_param_update);
+
+		} else {
+			param_get(_analog_param_handles.v_div, &_analog_params.v_div);
+			param_get(_analog_param_handles.a_per_v, &_analog_params.a_per_v);
+			param_get(_analog_param_handles.adc_channel, &_analog_params.adc_channel);
+		}
+
+		param_get(_analog_param_handles.cnt_v_volt, &_analog_params.cnt_v_volt);
+		param_get(_analog_param_handles.cnt_v_curr, &_analog_params.cnt_v_curr);
+		param_get(_analog_param_handles.v_offs_cur, &_analog_params.v_offs_cur);
+
+
+		if (_analog_params.cnt_v_volt < 0.0f) {
+			// TODO: This magic constant was hardcoded into sensors.cpp before I did the refactor. I don't know
+			//  what the best way is to make it not a magic number.
+			_analog_params.cnt_v_volt = DEFAULT_VOLTS_PER_COUNT;
```

Comment:
It's slightly different in current master.

https://github.com/PX4/Firmware/blob/bd98a4eebbee750a4db007b03bcd1e40896cb5d5/src/modules/battery_status/parameters.cpp#L71-L73